plugins/synaptics-mst/fu-synaptics-mst-device.c

```diff
@@ -871,14 +997,21 @@ fu_synaptics_mst_device_write_firmware (FuDevice *device,
 			g_prefix_error (error, "Firmware update failed: ");
 			return FALSE;
 		}
-	} else {
-		if (!fu_synaptics_mst_device_update_tesla_leaf_firmware (self,
+		break;
+	case FU_SYNAPTICS_MST_FAMILY_CAYENNE:
+	case FU_SYNAPTICS_MST_FAMILY_SPYDER:
+	g_debug(" === Update cayenne/spyder firmware === \n");
+		if (!fu_synaptics_mst_device_update_cayenne_firmware (self,
 									 payload_len,
 									 payload_data,
 									 error)) {
 			g_prefix_error (error, "Firmware update failed: ");
 			return FALSE;
 		}
+		break;
+	default:
+		g_prefix_error (error, "Unsupport chip family: ");
```

Comment:
you need to use `g_set_error()` if there's no existing error set.

Reply:
Hi hughsie,
Many thanks for your reviews.
Now I have done the modifications based on your reviews and suggestions.
I have git push the commit to https://github.com/apolloling/fwupd
What do I need to continue the PR ?